src/rgw/rgw_sal_motr.cc

```diff
@@ -2319,6 +2338,28 @@ int MotrObject::copy_object_same_zone(RGWObjectCtx& obj_ctx,
     *etag = etag_str;
   }
 
+  //Set object tags based on tagging-directive
+  struct req_state* s = static_cast<req_state*>(obj_ctx.get_private());
+  auto tagging_dir = s->info.env->get("HTTP_X_AMZ_TAGGING_DIRECTIVE");
+
+  bufferlist tags_bl;
+  if (tagging_dir) {
+    if (strcasecmp(tagging_dir, "COPY") == 0) {
+      rc = read_op->get_attr(dpp, RGW_ATTR_TAGS, tags_bl, y);
+      if (rc < 0) {
+        ldpp_dout(dpp, 20) <<__func__<< "ERROR: read op for object tags failed rc=" << rc << dendl;
+        return rc;
+      }
+    } else if (strcasecmp(tagging_dir, "REPLACE") == 0) {
+      int r = parse_tags(dpp, tags_bl, s);
```

Comment:
please add a debug log saying 'parsing tag values for object : {obj name}'

Reply:
Added